### DISCUSSION TITLE
Variable stroke weight/width

### DISCUSSION THREAD

**Discussion by launch-daemon** [keiwando/toonsquid#675] [Ideas]
As vector brush editing is coming, I feel like we could also have a feature to allow us to freely edit the width of the stroke at any point in the line:
<img width="609" alt="Screenshot of Affinity Designer 2, showing a line with varying width." src="https://github.com/user-attachments/assets/863936e4-6e6d-4141-922f-a3626547fe96">

**Comment by TempleLionStudios** [comment]
This would be a gamechanger!

Being able to import a character (we need a character library across projects!) and scale them to the scene, and adjust stroke size to fit the scaling! 👌🏽

**Comment by Nayelianne** [comment]
Just updated to 2.0 in the hopes that we now had this feature for Vector Stroke Brushes (which finally addressed the issues I had with the unstable thickness in the vector shape brushes), but couldn't find a way to change thickness for certain parts without changing it for the entire stroke, so came here to suggest this also. 

🥺 If we could just go over the part of the line we want to thicken/make thinner with a brush (kinda like in both SAI and Clip Studio Paint), it would be the best approach in my opinion (instead of having to slowly fiddle with a ton of points and fight to have a smooth line like in the case of the vector shape brush)

Reference video of what I mean: 
Clip Studio: https://www.youtube.com/watch?v=yAqKJTJdFtA
Paint Tool SAI: https://youtu.be/ONQVRR9635Y?si=6w71fvYoqchvjC9K&t=34

**Comment by StarParad0x** [comment]
This is a feature that certainly would improve working with vector lineart! Agreed 100%

**Comment by DWrightAnimation** [comment]
Frankly, I wish we can have both this version of vectors and the vector shape brushes. When doing motion tweening, it's hard to have strokes that vary in size throughout because we have to move every little point in the vector shape, but if we could do it like we can with regular vector strokes, it'd be easier.
